### PR TITLE
audit-proxy image for dynamic audit e2e test

### PR DESCRIPTION
**Pull request by pbarker**
/kind feature

Adds an audit-proxy image for use with the dynamic audit e2e test https://github.com/kubernetes/kubernetes/pull/70036

This needs to be merged separately so it can be pushed and added to the test. Issue https://github.com/kubernetes/kubernetes/pull/70036

```release-note
NONE
```

### CODE REVIEW

**Comment by neolit123** [comment]
@pbarker please add a release note.

/ok-to-test

what is the priority of this feature? adding longterm.
/priority important-longterm

File: test/images/audit-proxy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM scratch
+COPY audit-proxy /
+ENTRYPOINT ["/audit-proxy"]
```
Comment (mkumatag) [inline test/images/audit-proxy/Dockerfile]: minor: add newline

File: test/images/audit-proxy/VERSION [hunk 2]
```diff
@@ -0,0 +1 @@
+1.0
```
Comment (mkumatag) [inline test/images/audit-proxy/VERSION]: minor: add newline

**Review by mkumatag — CHANGES_REQUESTED** [review]
minors comments otherwise LGTM.

File: test/images/audit-proxy/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2018 The Kubernetes Authors.
```
Comment (ncdc) [inline test/images/audit-proxy/Dockerfile]: Should this be 2019?

File: test/images/audit-proxy/Makefile [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2018 The Kubernetes Authors.
```
Comment (ncdc) [inline test/images/audit-proxy/Makefile]: Should this be 2019?

File: test/images/audit-proxy/main.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	decoder = audit.Codecs.UniversalDecoder(auditv1.SchemeGroupVersion)
+
+	http.HandleFunc("/", handler)
+	log.Fatal(http.ListenAndServe(":8080", nil))
+}
+
+func handler(w http.ResponseWriter, req *http.Request) {
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		log.Fatalf("could not read request body: %v", err)
```
Comment (ncdc) [inline test/images/audit-proxy/main.go]: Do you want to terminate the process if it gets an error reading the request body, or should it instead return a 500 internal server error to the client?

File: test/images/audit-proxy/main.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	}
+	el := &auditv1.EventList{}
+
+	if err := runtime.DecodeInto(decoder, body, el); err != nil {
+		log.Fatalf("failed decoding buf: %b, apiVersion: %s", body, auditv1.SchemeGroupVersion)
```
Comment (ncdc) [inline test/images/audit-proxy/main.go]: Same terminate vs 500 question

File: test/images/audit-proxy/main.go [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+	}
+	defer req.Body.Close()
+
+	// write events to stdout
+	for _, event := range el.Items {
+		err := encoder.Encode(&event, os.Stdout)
+		if err != nil {
+			log.Fatalf("could not encode audit event: %v", err)
```
Comment (ncdc) [inline test/images/audit-proxy/main.go]: Same terminate vs 500 question

File: test/images/audit-proxy/main.go [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+		}
+	}
+	w.WriteHeader(http.StatusOK)
+	return
```
Comment (ncdc) [inline test/images/audit-proxy/main.go]: Unnecessary `return`?

File: test/images/audit-proxy/Dockerfile (hunk 3 above)
Comment (pbarker) [reply test/images/audit-proxy/Dockerfile]: 👍 this ones been hanging out for awhile

File: test/images/audit-proxy/main.go (hunk 5 above)
Comment (pbarker) [reply test/images/audit-proxy/main.go]: It would probably be most obvious to log and return a 500, updating

**Comment by nikhita** [comment]
@pbarker could you add a release note if needed? Thanks! :)

**Comment by ixdy** [comment]
image has been built and pushed by @ixdy-bot.

would still love to see automation around this.

**Pull request merged**